app/models/manageiq/providers/ibm_power_hmc/infra_manager/event_target_parser.rb

```diff
@@ -20,35 +20,35 @@ def parse
     case ems_event.event_type
     when "MODIFY_URI", "ADD_URI", "DELETE_URI" # Damien: INVALID_URI?
       $ibm_power_hmc_log.info("#{self.class}##{__method__} #{ems_event.event_type} #{ems_event.full_data}")
-      elems = elem(raw_event[:data])
-      case elems[:type]
+      elems = raw_event[:data].match(/^.+\/rest\/api\/uom\/(ManagedSystem\/(?<manager_uuid>.+)\/)?(?<type>.+)\/(?<uuid>.+)$/).named_captures
```

Comment:
Considering all of the escaped `/`, I'm surprised rubocop didn't suggest to use %r{}, so I will :)
```suggestion
      elems = raw_event[:data].match(%r{^.+/rest/api/uom/(ManagedSystem/(?<manager_uuid>.+)/)?(?<type>.+)/(?<uuid>.+)$}).named_captures
```

Reply:
You might also want to do `&.named_captures`, just in case it doesn't match, and then have a exception or something if elems ends up being nil.

Reply:
...or whatever the old code did - probably was just hashes with nils - not sure, but should be something covered by tests.